### ISSUE TITLE
Notes

### ISSUE THREAD

**Issue by AntonC9018** [biglevoozy/ASDC---Tsurtsurica#1]
![image](https://github.com/biglevoozy/ASDC---Tsurtsurica/assets/39066090/21057e87-2516-471b-9fda-594a24eb9e15)
`.gitignore`? Посмотри видосик. Не комить эти файлы. На паре покажу как удалить, если уже закомитил. Покажу и как выпилить из истории мб, потому что ничего из этого не надо в репозитории.

**Comment by AntonC9018** [comment]
https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/5c7f7ce2b34a49ce386f80d557bd20c3bccf57f6/Library/ListFncts.cpp#L62-L63
Нет надобности различать. Там nullptr по умолчанию.

**Comment by AntonC9018** [comment]
https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/5c7f7ce2b34a49ce386f80d557bd20c3bccf57f6/Library/HumanFncts.cpp#L54
Сделай проверку на енум значение, зачем это?

**Comment by AntonC9018** [comment]
Это сделал сам
https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/5c7f7ce2b34a49ce386f80d557bd20c3bccf57f6/Lab_6_Hash_Tables/main.cpp#L37-L54

А это скопировал?
https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/5c7f7ce2b34a49ce386f80d557bd20c3bccf57f6/Lab_6_Hash_Tables/main.cpp#L87-L95

**Comment by AntonC9018** [comment]
https://github.com/biglevoozy/ASDC---Tsurtsurica/tree/main/Library
Я понимаю, что ты восхитился структурой другого студента. Это не есть плохо. Но зачем файлы переименовывать сокращениями?)

**Comment by AntonC9018** [comment]
https://github.com/biglevoozy/ASDC---Tsurtsurica/tree/main/Lab_5_Queue
Пустое?

**Comment by AntonC9018** [comment]
https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/5c7f7ce2b34a49ce386f80d557bd20c3bccf57f6/Lab_3_DoubleLinkedList/main.cpp#L177
Тут если есть цикл, оно уйдет при подсчете в бесконечный цикл.
Не используй слово claim для этого, для этой идеи assert -- стандартный термин. Claim не понятно что значит сам по себе.

**Comment by AntonC9018** [comment]
Лабы с замером (2 и 7) будешь делать?

**Comment by biglevoozy** [comment]
> https://github.com/biglevoozy/ASDC---Tsurtsurica/tree/main/Lab_5_Queue Пустое?

благодарю, что сказали. видимо, когда разбирался с коммитами и .gitignore, что-то не то пошло. заглянув в проект, всё есть. попробую закомитить ещё разок

**Comment by biglevoozy** [comment]
> Это сделал сам
> 
> https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/5c7f7ce2b34a49ce386f80d557bd20c3bccf57f6/Lab_6_Hash_Tables/main.cpp#L37-L54
> 
> А это скопировал?
> 
> https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/5c7f7ce2b34a49ce386f80d557bd20c3bccf57f6/Lab_6_Hash_Tables/main.cpp#L87-L95

да, вторую часть кода скопировал. на одной из пар, мы вместе этот участок кода разбирали, когда вы подсказывали, как это можно и нужно интерпретировать, потому, каюсь, что скопировал.

**Comment by biglevoozy** [comment]
> https://github.com/biglevoozy/ASDC---Tsurtsurica/tree/main/Library Я понимаю, что ты восхитился структурой другого студента. Это не есть плохо. Но зачем файлы переименовывать сокращениями?)

поскольку стараюсь писать сам, с объяснениями от другого студента, по своей привычке так называю файлы.

**Comment by biglevoozy** [comment]
> Лабы с замером (2 и 7) будешь делать?

да, постараюсь сделать.

**Comment by AntonC9018** [comment]
https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/76aa49cbd11793b8302e8db88ffdbed32eecfca1/Lab_7_Sorting_Algorithms/README.md?plain=1#L33-L39
Ну раз время везде 0, что-то явно не то. Ты на основе таких данных и не сможешь нормальный анализ провести.

**Comment by AntonC9018** [comment]
Просмотрю еще более углубленно завтра

**Comment by biglevoozy** [comment]
> https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/76aa49cbd11793b8302e8db88ffdbed32eecfca1/Lab_7_Sorting_Algorithms/README.md?plain=1#L33-L39
> 
> 
> Ну раз время везде 0, что-то явно не то. Ты на основе таких данных и не сможешь нормальный анализ провести.

исправил проблему, у меня не считывался весь файл, от того и всё происходило слишком быстро. теперь в анализе весь массив данных и время более не менее доступно для сравнения.

**Comment by AntonC9018** [comment]
Допилишь анализ и дофиксишь эти проблемы, будет 10

**Comment by AntonC9018** [comment]
https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Library/HumanFncts.cpp#L54-L67
Зачем `std::string`? Ты же просто печатаешь это, зачем память выделять? `std::string_view` чтобы смотрел в статическую память.

https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Library/HumanFncts.cpp#L71
Удали откоменченный код.

https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Library/HumanFncts.cpp#L165
А остальное что? Просто игнорится? Если ты передашь `1dsffds0`, он это запарсит как `1`. Это действительно желаемый результат?

https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Library/ListFncts.cpp#L30
Только щас посмторел, что у тебя max_size для строки идет 999. Если так много символов позволяешь, копировать в стек такое будет дорого. Передавай референсом или указателем если можешь. Но вообще правильнее всего для такого огромного объекта было бы вернуть указатель, и туда вписать инфу, чтобы не копировать по возможности. Или сделай меньше строку. 999 слишком много имо.

https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Lab_7_Sorting_Algorithms/main.cpp#L43
span

В доки с анализом вставь конкретно какие результаты получил. Скопируй из консоли сколько секунд занял алгоритм, сколько операций сравнения, т.п., для разных размеров массива. И из этого выведи сложность алгоритма. Пока 9, допилишь это, и пофиксишь заметки, будет 10

**Comment by biglevoozy** [comment]
> https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Library/HumanFncts.cpp#L54-L67
> 
> 
> Зачем `std::string`? Ты же просто печатаешь это, зачем память выделять? `std::string_view` чтобы смотрел в статическую память.
> https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Library/HumanFncts.cpp#L71
> 
> 
> Удали откоменченный код.
> https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Library/HumanFncts.cpp#L165
> 
> 
> А остальное что? Просто игнорится? Если ты передашь `1dsffds0`, он это запарсит как `1`. Это действительно желаемый результат?
> https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Library/ListFncts.cpp#L30
> 
> 
> Только щас посмторел, что у тебя max_size для строки идет 999. Если так много символов позволяешь, копировать в стек такое будет дорого. Передавай референсом или указателем если можешь. Но вообще правильнее всего для такого огромного объекта было бы вернуть указатель, и туда вписать инфу, чтобы не копировать по возможности. Или сделай меньше строку. 999 слишком много имо.
> https://github.com/biglevoozy/ASDC---Tsurtsurica/blob/1e5bbb62d2fccbcc8f5bad9bbee928d0e3d7453a/Lab_7_Sorting_Algorithms/main.cpp#L43
> 
> 
> span
> В доки с анализом вставь конкретно какие результаты получил. Скопируй из консоли сколько секунд занял алгоритм, сколько операций сравнения, т.п., для разных размеров массива. И из этого выведи сложность алгоритма. Пока 9, допилишь это, и пофиксишь заметки, будет 10

все помеченные ошибки в коде и замечания подправил
